### PR TITLE
feat: Add tests and add feature to update daemonset to have selector

### PR DESCRIPTION
**Pull request by rivToadd**
**Description:** <Describe what has changed.>

add feature to update daemonset to have selector field

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #2605 

**Testing:** <Describe what testing was performed and which tests were added.>
wrote unit tests for said feature

ran test on local kind cluster:

yaml spec used to deploy: 
```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: logs
  labels:
    app: logs
spec:
  mode: daemonset
  hostNetwork: true
  config: |
    receivers:
      jaeger:
        protocols:
          grpc:
    processors:

    exporters:
      debug:
        verbosity: detailed

    service:
      pipelines:
        traces:
          receivers: [jaeger]
          processors: []
          exporters: [debug]
  priorityClassName: system-cluster-critical
  resources:
    limits:
      memory: 128Mi
    requests:
      cpu: 10m
      memory: 128Mi
  tolerations:
  - operator: Exists
  volumeMounts:
  - mountPath: /var/log
    name: varlog
    readOnly: true
  volumes:
  - hostPath:
      path: /var/log
    name: varlog

---

apiVersion: autoscaling.k8s.io/v1
kind: VerticalPodAutoscaler
metadata:
  name: logs-collector
spec:
  resourcePolicy:
    containerPolicies:
    - containerName: '*'
      controlledResources:
      - cpu
      - memory
  targetRef:
    apiVersion: opentelemetry.io/v1alpha1
    kind: OpenTelemetryCollector
    name: logs
  updatePolicy:
    updateMode: Auto
```
Logs that I saw on VPA
```
I0226 22:37:30.086614       1 matcher.go:68] Let's choose from 1 configs for pod default/logs-collector-%
I0226 22:37:30.086717       1 recommendation_provider.go:91] updating requirements for pod logs-collector-%.
I0226 22:37:30.086839       1 server.go:112] Sending patches: [{add /spec/containers/0/resources/requests/memory 262144k} {add /spec/containers/0/resources/requests/cpu 25m} {add /spec/containers/0/resources/limits/memory 262144k} {add /metadata/annotations/vpaUpdates Pod resources updated by logs-collector: container 0: memory request, cpu request, memory limit} {add /metadata/annotations/vpaObservedContainers otc-container}]
```
resulting in new resource limits on collector pod
```
   Limits:
      memory:  262144k
    Requests:
      cpu:     25m
      memory:  262144k
```

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: internal/status/collector/collector.go [hunk 1]
```diff
@@ -35,8 +35,10 @@ func UpdateCollectorStatus(ctx context.Context, cli client.Client, changed *v1al
 		// a version is not set, otherwise let the upgrade mechanism take care of it!
 		changed.Status.Version = version.OpenTelemetryCollector()
 	}
+
 	mode := changed.Spec.Mode
-	if mode != v1alpha1.ModeDeployment && mode != v1alpha1.ModeStatefulSet {
+
+	if mode != v1alpha1.ModeDeployment && mode != v1alpha1.ModeStatefulSet && mode != v1alpha1.ModeDaemonSet {
```
Comment (swiatekm-sumo) [inline internal/status/collector/collector.go:41]: These are the only possible modes, so we can simply remove this condition.

**Review by swiatekm-sumo — COMMENTED** [review]
LGTM overall. Have you tested if the VPA works correctly after this change?

File: internal/status/collector/collector.go (hunk 1 above)
Comment (jaronoff97) [reply internal/status/collector/collector.go:41]: or we could just do `mode == v1alpha1.Sidecar`

**Comment by rivToadd** [comment]
@swiatekm-sumo  I've updated the PR description to document the local testing that I did.

**Review by swiatekm-sumo — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**